basiclti/basiclti-blis/src/java/org/sakaiproject/blti/ServiceServlet.java

```diff
@@ -641,6 +642,14 @@ protected void processRoster(HttpServletRequest request, HttpServletResponse res
 			Set<Member> members = site.getMembers();
 			Map<String, String> roleMap = SakaiBLTIUtil.convertRoleMapPropToMap(roleMapProp);
 			for (Member member : members ) {
+				User user;
+				try {
+					user = UserDirectoryService.getUser(member.getUserId());
```

Comment:
Now that I think of it, we have noticed in the past that lti roster updates are a little slow for large sites. This is probably why.